### ISSUE TITLE
Only first constant value of projected type is included in generated query

### ISSUE THREAD

**Issue by Panuf** [dotnet/efcore#28514]
When selecting query result as plain object and including constant values for some properties only first constant value for each type gets included in generated query, ie.

```c#
databaseContext.Items.Select(item => new Projection()
{
    Name1 = item.Name,
    Name2 = String.Empty,
    Name3 = String.Empty,
    Name4 = "a",
    Id1 = item.Id,
    Id2 = Guid.Empty,
    Id3 = Guid.Empty,
    Id4 = Guid.Empty,
    Number1 = 0,
    Number2 = 0,
    Number3 = 3,
    Number4 = 4
}
```

gets translated into 

```sql
SELECT [i].[Name] AS [Name1], N'' AS [Name2], N'a' AS [Name4], [i].[Id] AS [Id1], '00000000-0000-0000-0000-000000000000' AS [Id2], 0 AS [Number1], 3 AS [Number3], 4 AS [Number4]
FROM [Items] AS [i]
```

Name3, Id2-4 and Number2 is not included in generated query. 

The reason why I need two const values of each type (in my case those are two guids) is that I have two queries which I need to concact and in one query some properties are not used but still need to be included for concat to work. 

I am using EntityFrameworkCore and SqlServer both in version 6.0.7.
[Sample.zip](https://github.com/dotnet/efcore/files/9188712/Sample.zip)

**Comment by smitpatel** [comment]
Projecting same constant only once is by-design. What is the error message here?

**Comment by Panuf** [comment]
There is no error message with that query. It would result in error when union/concat with different query as underling SQL would have different number of columns.

**Comment by stevendarby** [comment]
When I tried it with a Concat, it no longer optimised, instead it selected the constants multiple times and there was no SQL error.

**Comment by Panuf** [comment]
Yes. Of course I was analysing queries in separation, which lead me to wrong conclusion. I am sorry for that I will create separate ticket with actual cause of the problem.
